aten/src/TH/generic/THTensorEvenMoreMath.cpp

```diff
@@ -831,15 +831,15 @@ void THTensor_(rshift)(THTensor *r_, THTensor *t, scalar_t value)
 #if defined(TH_REAL_IS_BYTE)
         rp[i] = ((scalar_t) tp[i]) >> value;
 #else
-        rp[i] = ((ureal) tp[i]) >> value;
+        rp[i] = ((accreal) tp[i]) >> value;
```

Comment:
why is doing this in terms of accreal the correct thing to do as opposed to scalar_t?